spec/decanter/parser/json_parser_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe 'JsonParser' do
+
+  let(:name) { :foo }
+
+  let(:parser) { Decanter::Parser::JsonParser }
+
+  describe '#parse' do
+    it 'parses string value and returns a parsed JSON' do
+      expect(parser.parse(name, '{"key": "value"}')).to match({name => {"key" => "value"}})
+    end
+
+    context 'with empty string' do
+      it 'returns nil' do
+        expect(parser.parse(name, '')).to match({name => nil})
+      end
+    end
+
+    context 'with nil' do
+      it 'returns nil' do
+        expect(parser.parse(name, nil)).to match({name => nil})
+      end
+    end
+  end
```

Comment:
As mentioned [here](https://github.com/LaunchPadLab/decanter/pull/140/files#r1287155575), I'd love to see a descriptive error thrown if the string can't be parsed as valid JSON. I think that would also warrant a test case here.

Reply:
Might also be worth adding test cases to ensure a JSON array string (i.e. `'['investment', 'early_retirement', 'saving']'`) and an object (`'{ "first_name": "Daniel", "last_name": "Schadd" }'` can both be handled.